[
    {
        "user": "UCQB6S222",
        "type": "message",
        "ts": "1684249376.365959",
        "client_msg_id": "e6cb65f2-3b5b-4c67-859c-6e18c416e204",
        "text": "I just started using betfairviz today. I am very frequently getting an exception and I wondered if anyone knows how I can avoid it. It looks like this:\n```File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/iostream.py\", line 182, in advance\n    assert 0 &lt; size &lt;= self._size\n  AssertionError```",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g953ffd5e3b7",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/5813be539113b05ece8212c2a06d006a.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "George",
            "real_name": "George",
            "display_name": "George",
            "team": "T4G9NBD2M",
            "name": "gd",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "reply_count": 20,
        "reply_users_count": 2,
        "latest_reply": "1684731657.165579",
        "reply_users": [
            "UBS7QANF3",
            "UCQB6S222"
        ],
        "replies": [
            {
                "user": "UBS7QANF3",
                "ts": "1684249429.218119"
            },
            {
                "user": "UCQB6S222",
                "ts": "1684249496.563659"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1684249591.448539"
            },
            {
                "user": "UCQB6S222",
                "ts": "1684249596.815579"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1684249627.557739"
            },
            {
                "user": "UCQB6S222",
                "ts": "1684249701.644989"
            },
            {
                "user": "UCQB6S222",
                "ts": "1684249747.895949"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1684249775.932359"
            },
            {
                "user": "UCQB6S222",
                "ts": "1684249784.218849"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1684249886.719419"
            },
            {
                "user": "UCQB6S222",
                "ts": "1684249938.075769"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1684250002.912219"
            },
            {
                "user": "UCQB6S222",
                "ts": "1684250004.348869"
            },
            {
                "user": "UCQB6S222",
                "ts": "1684250029.455009"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1684250789.574059"
            },
            {
                "user": "UCQB6S222",
                "ts": "1684251082.190079"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1684251162.250229"
            },
            {
                "user": "UCQB6S222",
                "ts": "1684251453.875759"
            },
            {
                "user": "UCQB6S222",
                "ts": "1684703594.051569"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1684731657.165579"
            }
        ],
        "is_locked": false,
        "subscribed": false,
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "m8CC",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "I just started using betfairviz today. I am very frequently getting an exception and I wondered if anyone knows how I can avoid it. It looks like this:\n"
                            }
                        ]
                    },
                    {
                        "type": "rich_text_preformatted",
                        "elements": [
                            {
                                "type": "text",
                                "text": "File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/iostream.py\", line 182, in advance\n    assert 0 < size <= self._size\n  AssertionError"
                            }
                        ],
                        "border": 0
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1684249429.218119",
        "client_msg_id": "44DECAC4-E5E3-44C2-9611-EEF2C2B3DDB2",
        "text": "What’s the full stack trace",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "WCfm",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "What’s"
                            },
                            {
                                "type": "text",
                                "text": " the full stack trace"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UCQB6S222",
        "type": "message",
        "ts": "1684249496.563659",
        "client_msg_id": "d764fef6-c0c4-4a15-9113-26dd40d21106",
        "text": "```Exception in callback functools.partial(&lt;function ZMQStream._update_handler.&lt;locals&gt;.&lt;lambda&gt; at 0x107b25160&gt;)\n  Traceback (most recent call last):\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/ioloop.py\", line 738, in _run_callback\n    ret = callback()\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/zmq\/eventloop\/zmqstream.py\", line 718, in &lt;lambda&gt;\n    self.io_loop.add_callback(lambda: self._handle_events(self.socket, 0))\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/zmq\/eventloop\/zmqstream.py\", line 634, in _handle_events\n    self._handle_recv()\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/zmq\/eventloop\/zmqstream.py\", line 663, in _handle_recv\n    self._run_callback(callback, msg)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/zmq\/eventloop\/zmqstream.py\", line 584, in _run_callback\n    f = callback(*args, **kwargs)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/zmq\/eventloop\/zmqstream.py\", line 308, in stream_callback\n    return callback(self, msg)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/notebook\/services\/kernels\/handlers.py\", line 572, in _on_zmq_reply\n    super()._on_zmq_reply(stream, msg)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/notebook\/base\/zmqhandlers.py\", line 256, in _on_zmq_reply\n    self.write_message(msg, binary=isinstance(msg, bytes))\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/websocket.py\", line 334, in write_message\n    return self.ws_connection.write_message(message, binary=binary)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/websocket.py\", line 1081, in write_message\n    fut = self._write_frame(True, opcode, message, flags=flags)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/websocket.py\", line 1056, in _write_frame\n    return self.stream.write(frame)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/iostream.py\", line 539, in write\n    self._handle_write()\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/iostream.py\", line 965, in _handle_write\n    self._write_buffer.advance(num_bytes)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/iostream.py\", line 182, in advance\n    assert 0 &lt; size &lt;= self._size\n  AssertionError```",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g953ffd5e3b7",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/5813be539113b05ece8212c2a06d006a.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "George",
            "real_name": "George",
            "display_name": "George",
            "team": "T4G9NBD2M",
            "name": "gd",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "6OOU",
                "elements": [
                    {
                        "type": "rich_text_preformatted",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Exception in callback functools.partial(<function ZMQStream._update_handler.<locals>.<lambda> at 0x107b25160>)\n  Traceback (most recent call last):\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/ioloop.py\", line 738, in _run_callback\n    ret = callback()\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/zmq\/eventloop\/zmqstream.py\", line 718, in <lambda>\n    self.io_loop.add_callback(lambda: self._handle_events(self.socket, 0))\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/zmq\/eventloop\/zmqstream.py\", line 634, in _handle_events\n    self._handle_recv()\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/zmq\/eventloop\/zmqstream.py\", line 663, in _handle_recv\n    self._run_callback(callback, msg)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/zmq\/eventloop\/zmqstream.py\", line 584, in _run_callback\n    f = callback(*args, **kwargs)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/zmq\/eventloop\/zmqstream.py\", line 308, in stream_callback\n    return callback(self, msg)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/notebook\/services\/kernels\/handlers.py\", line 572, in _on_zmq_reply\n    super()._on_zmq_reply(stream, msg)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/notebook\/base\/zmqhandlers.py\", line 256, in _on_zmq_reply\n    self.write_message(msg, binary=isinstance(msg, bytes))\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/websocket.py\", line 334, in write_message\n    return self.ws_connection.write_message(message, binary=binary)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/websocket.py\", line 1081, in write_message\n    fut = self._write_frame(True, opcode, message, flags=flags)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/websocket.py\", line 1056, in _write_frame\n    return self.stream.write(frame)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/iostream.py\", line 539, in write\n    self._handle_write()\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/iostream.py\", line 965, in _handle_write\n    self._write_buffer.advance(num_bytes)\n   File \"\/Users\/george\/Library\/Python\/3.9\/lib\/python\/site-packages\/tornado\/iostream.py\", line 182, in advance\n    assert 0 < size <= self._size\n  AssertionError"
                            }
                        ],
                        "border": 0
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1684249591.448539",
        "client_msg_id": "203DF424-69E3-4F84-8312-8530622FD5B2",
        "text": "You’re running this in a jupyter notebook?",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "HwBQz",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "You’re"
                            },
                            {
                                "type": "text",
                                "text": " running this in a jupyter notebook?"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UCQB6S222",
        "type": "message",
        "ts": "1684249596.815579",
        "client_msg_id": "c0c7d7ea-9e27-4c37-8f39-5bec6127ac65",
        "text": "yes",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g953ffd5e3b7",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/5813be539113b05ece8212c2a06d006a.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "George",
            "real_name": "George",
            "display_name": "George",
            "team": "T4G9NBD2M",
            "name": "gd",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "MB8",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "yes"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1684249627.557739",
        "client_msg_id": "2ACF461F-872E-449B-8C8F-F578DD2D540B",
        "text": "It looks to me like it’s some jupyter error unrelated to betfairviz but I could be wrong",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "Gf8W0",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "It looks to me like "
                            },
                            {
                                "type": "text",
                                "text": "it’s"
                            },
                            {
                                "type": "text",
                                "text": " some jupyter error unrelated to betfairviz but I could be wrong"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UCQB6S222",
        "type": "message",
        "ts": "1684249701.644989",
        "client_msg_id": "db8df6b1-272a-4b61-a15d-16e8bd4f1d96",
        "text": "it only happens when I use `create_dashboard` and use the forward arrow\ni should add it only happens about 10% of the time",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g953ffd5e3b7",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/5813be539113b05ece8212c2a06d006a.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "George",
            "real_name": "George",
            "display_name": "George",
            "team": "T4G9NBD2M",
            "name": "gd",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "buX1d",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "it only happens when I use "
                            },
                            {
                                "type": "text",
                                "text": "create_dashboard",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "text",
                                "text": " and use the forward arrow\ni should add it only happens about 10% of the time"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UCQB6S222",
        "type": "message",
        "ts": "1684249747.895949",
        "client_msg_id": "4d42f25b-ed83-42fa-a51b-753a0cb2a54c",
        "text": "it definitely looks like a jupyter error but i'm wondering if anyone can suggest maybe a different version of a package that i should upgrade to for example?",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g953ffd5e3b7",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/5813be539113b05ece8212c2a06d006a.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "George",
            "real_name": "George",
            "display_name": "George",
            "team": "T4G9NBD2M",
            "name": "gd",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "p\/Zg5",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "it definitely looks like a jupyter error but i'm wondering if anyone can suggest maybe a different version of a package that i should upgrade to for example?"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1684249775.932359",
        "client_msg_id": "CE04A211-958F-4A8C-95BD-0E43E6631C0F",
        "text": "And when the error occurs does the dashboard break or is it still usable?",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "NpHH",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "And when the error occurs does the dashboard break or is it still usable?"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UCQB6S222",
        "type": "message",
        "ts": "1684249784.218849",
        "client_msg_id": "af74fd05-682e-41b6-bd2f-e6498dc90511",
        "text": "it breaks",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g953ffd5e3b7",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/5813be539113b05ece8212c2a06d006a.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "George",
            "real_name": "George",
            "display_name": "George",
            "team": "T4G9NBD2M",
            "name": "gd",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "98lI",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "it breaks"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1684249886.719419",
        "client_msg_id": "20E4BC2C-515B-47A9-B031-DCC75806C061",
        "text": "I know you said it only happens about 10% of the time but is that randomly or is it deterministic when advancing to particular market books?",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "3Vdyu",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "I know you said it only happens about 10% of the time but is that randomly or is it deterministic when advancing to particular market books?"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UCQB6S222",
        "type": "message",
        "ts": "1684249938.075769",
        "client_msg_id": "37e66345-6d16-44df-a291-d426364cc97b",
        "text": "i'm not totally sure on that one",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g953ffd5e3b7",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/5813be539113b05ece8212c2a06d006a.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "George",
            "real_name": "George",
            "display_name": "George",
            "team": "T4G9NBD2M",
            "name": "gd",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "pAo8",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "i'm not totally sure on that one"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1684250002.912219",
        "client_msg_id": "4C45CFA8-45C6-4832-AF33-8316A2A3C6EB",
        "text": "Ok well I’ll have a play and see whether I can make it happen myself. Otherwise let me know if you get any more insights ",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "7x7L",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Ok well "
                            },
                            {
                                "type": "text",
                                "text": "I’ll"
                            },
                            {
                                "type": "text",
                                "text": " have a play and see whether I can make it happen myself"
                            },
                            {
                                "type": "text",
                                "text": "."
                            },
                            {
                                "type": "text",
                                "text": " Otherwise let me know if you get any more insights "
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UCQB6S222",
        "type": "message",
        "ts": "1684250004.348869",
        "client_msg_id": "0e285544-3d82-4812-a31f-fd82f4079d8b",
        "text": "i think it's random",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g953ffd5e3b7",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/5813be539113b05ece8212c2a06d006a.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "George",
            "real_name": "George",
            "display_name": "George",
            "team": "T4G9NBD2M",
            "name": "gd",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "RW5VY",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "i think it's random"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UCQB6S222",
        "type": "message",
        "ts": "1684250029.455009",
        "client_msg_id": "aa799488-d38b-44d6-bbd7-8b5c31c69bd2",
        "text": "i also think it happens more often if I click really fast",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g953ffd5e3b7",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/5813be539113b05ece8212c2a06d006a.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "George",
            "real_name": "George",
            "display_name": "George",
            "team": "T4G9NBD2M",
            "name": "gd",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "B6f",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "i also think it happens more often if I click really fast"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1684250789.574059",
        "client_msg_id": "1d9fa81c-c99a-4734-9368-1e1c5ee57c85",
        "text": "What version of tornado do you have installed?",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "quHD",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "What version of tornado do you have installed?"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UCQB6S222",
        "type": "message",
        "ts": "1684251082.190079",
        "client_msg_id": "5d586217-b943-4638-82c0-748d9e531e77",
        "text": "6.3.2",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g953ffd5e3b7",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/5813be539113b05ece8212c2a06d006a.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "George",
            "real_name": "George",
            "display_name": "George",
            "team": "T4G9NBD2M",
            "name": "gd",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "Z5z+q",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "6.3.2"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1684251162.250229",
        "client_msg_id": "864b9797-aabe-40ca-8b5b-9d9510f7275b",
        "text": "This thread: <https:\/\/discourse.jupyter.org\/t\/jupyter-notebook-zmq-message-arrived-on-closed-channel-error\/17869\/17> suggests you should try downgrading to 6.1",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "attachments": [
            {
                "from_url": "https:\/\/discourse.jupyter.org\/t\/jupyter-notebook-zmq-message-arrived-on-closed-channel-error\/17869\/17",
                "ts": 1678602822,
                "thumb_url": "https:\/\/global.discourse-cdn.com\/standard11\/uploads\/jupyter\/original\/1X\/50206ee532fcfae18c9359cfe327ed6844d0cf79.png",
                "thumb_width": 512,
                "thumb_height": 512,
                "service_icon": "https:\/\/global.discourse-cdn.com\/standard11\/uploads\/jupyter\/optimized\/1X\/31d4ec0f3f78ef012e9def937f37afe6ae68b8d5_2_180x180.png",
                "id": 1,
                "original_url": "https:\/\/discourse.jupyter.org\/t\/jupyter-notebook-zmq-message-arrived-on-closed-channel-error\/17869\/17",
                "fallback": "Jupyter Community Forum: Jupyter Notebook 'zmq message arrived on closed channel' Error",
                "text": "I finally found here… I also have same issue on my M1 Mac. My environment is, OS Ventura 13.2.1 VScode 1.76.1 conda 23.1.0 Tornado 6.2 jupyter_client 8.0.3 It means also same if I use jupyter from VScode. But I when I use exact same tools with same versions in my Linux on Parallels, it never happen. After downgraded my tornado to 6.1, It seems stable sofar… !#sample$ conda install tornado=6.1 and conda suggest below, I just installed these with OK. jupyter_client-7.3.4 | p...",
                "title": "Jupyter Notebook 'zmq message arrived on closed channel' Error",
                "title_link": "https:\/\/discourse.jupyter.org\/t\/jupyter-notebook-zmq-message-arrived-on-closed-channel-error\/17869\/17",
                "service_name": "Jupyter Community Forum"
            }
        ],
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "Id3",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "This thread: "
                            },
                            {
                                "type": "link",
                                "url": "https:\/\/discourse.jupyter.org\/t\/jupyter-notebook-zmq-message-arrived-on-closed-channel-error\/17869\/17"
                            },
                            {
                                "type": "text",
                                "text": " suggests you should try downgrading to 6.1"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UCQB6S222",
        "type": "message",
        "ts": "1684251453.875759",
        "client_msg_id": "c28a9783-cc46-4716-a711-e7b54d43c1cd",
        "text": "thanks, trying this now (had to downgrade jupyter client and server too)",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g953ffd5e3b7",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/5813be539113b05ece8212c2a06d006a.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "George",
            "real_name": "George",
            "display_name": "George",
            "team": "T4G9NBD2M",
            "name": "gd",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684249376.365959",
        "parent_user_id": "UCQB6S222",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "WHdm",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "thanks, trying this now (had to downgrade jupyter client and server too)"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UPMUFSGCR",
        "type": "message",
        "ts": "1684267282.474879",
        "client_msg_id": "1b9b7b24-abb4-4785-87b7-8fa23c803f2e",
        "text": "My logs are full of TOO_MANY_REQUESTS errors when calling listClearedOrders. How do people avoid that?",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "31c0bb5a442c",
            "image_72": "https:\/\/avatars.slack-edge.com\/2019-10-28\/812386967189_31c0bb5a442c5b8d2c61_72.png",
            "first_name": "Jon",
            "real_name": "Jon Jon Jon Jon Jon Jon Jon Jon",
            "display_name": "Jonjonjon",
            "team": "T4G9NBD2M",
            "name": "fcmisc",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684267282.474879",
        "reply_count": 4,
        "reply_users_count": 2,
        "latest_reply": "1685281109.846079",
        "reply_users": [
            "UUE6E1LA1",
            "UUJNW57D5"
        ],
        "replies": [
            {
                "user": "UUE6E1LA1",
                "ts": "1684269230.454499"
            },
            {
                "user": "UUJNW57D5",
                "ts": "1685210520.465859"
            },
            {
                "user": "UUE6E1LA1",
                "ts": "1685263643.978079"
            },
            {
                "user": "UUJNW57D5",
                "ts": "1685281109.846079"
            }
        ],
        "is_locked": false,
        "subscribed": true,
        "last_read": "1685281109.846079",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "oQ6ND",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "My logs are full of TOO_MANY_REQUESTS errors when calling listClearedOrders. How do people avoid that?"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UUE6E1LA1",
        "type": "message",
        "ts": "1684269230.454499",
        "client_msg_id": "e849f187-3002-41e8-ad4a-10e3e3a0e199",
        "text": "Probably not helpful to you as its not flumine related but I have a throttle on calls and call on a per market basis rather than using settled time intervals. The operation only gets settled orders and I don't need to get them in a rush as soon as market closes so I just force a minimum of 5 second delay between calls, and only once a market has been closed for at least 60 minutes. Then use the to\/from records and the moreAvailable flag to pull them down until I've got them all. I also make seperate calls for each of the  VOIDED, LAPSED, CANCELLED and SETTLED betStatus types.",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gf53cae45edd",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/f53cae45edd7f7b67bbe98819597612e.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0017-72.png",
            "first_name": "",
            "real_name": "D C",
            "display_name": "D C",
            "team": "T4G9NBD2M",
            "name": "oddsvantage",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1684267282.474879",
        "parent_user_id": "UPMUFSGCR",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "LGGb",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Probably not helpful to you as its not flumine related but I have a throttle on calls and call on a per market basis rather than using settled time intervals. The operation only gets settled orders and I don't need to get them in a rush as soon as market closes so I just force a minimum of 5 second delay between calls, and only once a market has been closed for at least 60 minutes. Then use the to\/from records and the moreAvailable flag to pull them down until I've got them all. I also make seperate calls for each of the  VOIDED, LAPSED, CANCELLED and SETTLED betStatus types."
                            }
                        ]
                    }
                ]
            }
        ]
    }
]